### PR TITLE
Move Partials Around

### LINKED ISSUE #124: buttons overflow
https://30.codelab.fun/accessing-all-nested-form-controls/

Screen Shot 2019-06-03 at 18 12 17

### PR DESCRIPTION
**Pull request by kirjs**
also closes #124

### CODE REVIEW

File: builder/config.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = (isDevMode = false) => Metalsmith(__dirname)
 	.metadata({
 		site: {
 			base: 'https://30.codelab.fun/',
+      title: '30 Seconds of angular',
```
Comment (fetis) [inline builder/config.js:25]: misaligned
```suggestion
     title: '30 Seconds of angular',
```

File: snippets/ng-content.md [hunk 2]
```diff
@@ -50,19 +50,19 @@ function template(useClass: string): string {
 }
 
 @Component({
-  selector: 'card',
+  selector: 'app-card',
```
Comment (fetis) [inline snippets/ng-content.md]: I think it's part from #193, can be removed

File: builder/partials/head.hbs [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
+    {{/if}}
 
-  <link rel="stylesheet" href="public/css/style.css">
+    {{> @partial-block }}
 
-  {{> @partial-block }}
+    <link rel="stylesheet" href="public/css/style.css">
 </head>
-
-{{!-- see body.hbs for end --}}
```
Comment (fetis) [inline builder/partials/head.hbs]: I would keep it

File: builder/partials/body.hbs [hunk 4]
```diff
@@ -1,14 +1,49 @@
 {{!-- see head.hbs for start --}}
 
-{{!-- don't indent, it'll break code examples --}}
-{{#if class }}
-<body class="{{ class }}">
-{{> @partial-block }}
-</body>
-{{ else }}
-<body>
+<body class="{{ bodyclass }}">
```
Comment (fetis) [inline builder/partials/body.hbs]: 😱

File: builder/partials/body.hbs [hunk 5]
```diff
@@ -1,14 +1,49 @@
...
+
+{{#unless hideHeader}}
+    <header class="mdl-card mdl-shadow--2dp card header-card">
+        <a href="/"><img src="public/img/logo_beta.png" style="height: 100px;" alt="30 seconds of angular"></a>
+
+        <div class="header-card__buttons">
+            <a class="twitter-follow-button" data-show-count="false" data-size="large"
+               href="https://twitter.com/30sec_angular">Follow @30sec_angular</a>
+
+            <!-- Place this tag where you want the button to render. -->
```
Comment (fetis) [inline builder/partials/body.hbs]:
```suggestion
```

File: builder/layouts/tag.hbs [hunk 6]
```diff
@@ -1,15 +1,14 @@
...
-
-  {{#each pagination.files}}
-    {{>snippet}}
-  {{/each}}
-
-{{/full-body}}
+{{#> head}}
+    <title>✨ {{capitalize tag}} - {{site.title}} </title>
+{{/head}}
+{{#> body}}
```
Comment (fetis) [inline builder/layouts/tag.hbs:5]: it reads better with indentation

```suggestion

{{#> body}}
```

File: builder/layouts/tag.hbs [hunk 7]
```diff
@@ -1,15 +1,14 @@
...
-    <h2>Tag: {{ tag }}</h2>
-  </div>
-
-  {{#each pagination.files}}
-    {{>snippet}}
-  {{/each}}
-
-{{/full-body}}
+{{#> head}}
+    <title>✨ {{capitalize tag}} - {{site.title}} </title>
```
Comment (fetis) [inline builder/layouts/tag.hbs]: we need to be consistent with `h2` below. I'm not sure the capitalization is the right choice, for tags like `Ng-content` it looks weird.

File: builder/layouts/snippet.hbs [hunk 8]
```diff
@@ -10,24 +9,25 @@
             height: 100%;
         }
     </style>
-    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:card" content="summary_large_image"/>
```
Comment (fetis) [inline builder/layouts/snippet.hbs]:
```suggestion
    <meta name="twitter:card" content="summary_large_image">
```

File: builder/layouts/snippet.hbs [hunk 9]
```diff
@@ -10,24 +9,25 @@
...
+--}}{{>snippet}}
 
-    {{#if (hasCode this)}}
-        <div class="mdl-card mdl-shadow--2dp card demo-card">
-            <iframe src="https://codelab.fun/angular/playground?code={{codeToBase64 this}}"></iframe>
-        </div>
-    {{/if}}
-
-{{/full-body}}
+{{#if (hasCode this)}}
```
Comment (fetis) [inline builder/layouts/snippet.hbs:28]: indentation?

File: builder/layouts/slides.hbs [hunk 10]
```diff
@@ -1,90 +1,89 @@
 {{#> head }}
-  <title>30 Seconds of Angular slides</title>
```
Comment (fetis) [inline builder/layouts/slides.hbs:2]: I would leave the title here

File: builder/layouts/slides.hbs [hunk 11]
```diff
@@ -1,90 +1,89 @@
...
+        .with-list ul {
+            margin-left: 130px;
+        }
+
+        .slide {
+            padding: 24px;
+        }
+    </style>
+{{/head}}
+{{#> body bodyclass="shower list" hideHeader=true }}
```
Comment (fetis) [inline builder/layouts/slides.hbs]: reads better with empty line
```suggestion

{{#> body bodyclass="shower list" hideHeader=true }}
```

File: builder/layouts/new.hbs [hunk 12]
```diff
@@ -1,27 +1,20 @@
...
-  <title>Add new snippet</title>
-  <style>
-    .layout {
-      display: flex;
-      flex-direction: column;
-    }
+{{#> head}}
+    <title>New Snippet - {{site.title}}</title>
+    <style>
+        body {
```
Comment (fetis) [inline builder/layouts/new.hbs:4]: if we insert some additional code in the `body` snippet this code will be broken. the previous solution under some conditions as well :(

File: builder/layouts/new.hbs [hunk 13]
```diff
@@ -1,27 +1,20 @@
...
-    .layout__form {
-      flex: 1;
-    }
-    .layout__form iframe {
-      width: 100%;
-      height: 100%;
-    }
-  </style>
-{{/head }}
+        iframe {
```
Comment (fetis) [inline builder/layouts/new.hbs]: it affects all Twitter buttons. We have one in the header.

**Review by fetis — CHANGES_REQUESTED** [review]

File: builder/layouts/new.hbs (hunk 12 above)
Comment (NothingEverHappens) [reply builder/layouts/new.hbs:4]: True, lol, we need screenshot diffing

File: builder/layouts/snippet.hbs (hunk 9 above)
Comment (NothingEverHappens) [reply builder/layouts/snippet.hbs:28]: I think comment above screwed it up, it's actually how it is formatted. 

We might get prettier or something.

File: builder/layouts/tag.hbs (hunk 7 above)
Comment (NothingEverHappens) [reply builder/layouts/tag.hbs]: dropped

File: builder/partials/body.hbs (hunk 4 above)
Comment (NothingEverHappens) [reply builder/partials/body.hbs]: 🙀

File: snippets/ng-content.md (hunk 2 above)
Comment (NothingEverHappens) [reply snippets/ng-content.md]: Should be gone after sync

**Review by fetis — APPROVED** [review]

**Pull request merged**